src/porepy/models/applications/terzaghi_model.py

```diff
@@ -0,0 +1,706 @@
+    ) -> float:
+        """Numerical consolidation coefficient.
+
+        Args:
+            displacement: Displacement solution of shape (sd.dim * sd.num_cells, ).
+            pressure: Pressure solution of shape (sd.num_cells, ).
+
+        Returns:
+            Numerical degree of consolidation.
+
+        """
+        sd = self.mdg.subdomains()[0]
+        h = self.params["height"]
+        m_v = self.confined_compressibility()
+        vertical_load = self.params["vertical_load"]
+        trace_u = self.displacement_trace(displacement, pressure)
+
+        u_inf = m_v * h * vertical_load
+        u_0 = 0
+        u = np.max(np.abs(trace_u[1 :: sd.dim]))
+
+        consol_deg = (u - u_0) / (u_inf - u_0)
+
+        return consol_deg
+
+    def displacement_trace(
+        self, displacement: np.ndarray, pressure: np.ndarray
+    ) -> np.ndarray:
+        """Project the displacement vector onto the faces.
+
+        Args:
+            displacement: displacement solution of shape (sd.dim * sd.num_cells, ).
+            pressure: pressure solution of shape (sd.num_cells, ).
+
+        Returns:
+            Trace of the displacement with shape (sd.dim * sd.num_faces, ).
+
+        """
+        # Rename arguments
+        u = displacement
+        p = pressure
+
+        # Discretization matrices
+        sd = self.mdg.subdomains()[0]
+        data = self.mdg.subdomain_data(sd)
+        discr = data[pp.DISCRETIZATION_MATRICES][self.mechanics_parameter_key]
+        bound_u_cell = discr["bound_displacement_cell"]
+        bound_u_face = discr["bound_displacement_face"]
+        bound_u_pressure = discr["bound_displacement_pressure"]
+
+        # Mechanical boundary values
+        bc_vals = data[pp.PARAMETERS][self.mechanics_parameter_key]["bc_values"]
+
+        # Compute trace of the displacement
+        trace_u = bound_u_cell * u + bound_u_face * bc_vals + bound_u_pressure * p
+
+        return trace_u
+
+    # -----> Helper methods
+    def plot_results(self) -> None:
+        """Plot the results"""
+        # Retrieve colormap from the tab20 pallete
+        cmap = mcolors.ListedColormap(
+            plt.cm.tab20.colors[: len(self.time_manager.schedule)]
+        )
+
+        # Pressure plot
+        self._pressure_plot(
+            folder="out/",
+            file_name="nondimensional_pressure",
+            file_extension=".pdf",
+            color_map=cmap,
+        )
+
+        # Degree of consolidation plot
+        self._consolidation_degree_plot(
+            folder="out/",
+            file_name="consolidation_degree",
+            file_extension=".pdf",
+            color_map=cmap,
+        )
+
+    def _pressure_plot(
+        self,
+        folder: str,
+        file_name: str,
+        file_extension: str,
+        color_map: mcolors.ListedColormap,
+    ) -> None:
+        """Plot nondimensional pressure profiles
+
+        Args:
+            folder: name of the folder to store the results e.g., "out/"
```

Comment:
Please add commas before e.g. 
I think @keileg is in favour of a full stop at the end of each line/sentence as well.

Reply:
I am very much in favor of full stops.

Reply:
Will do.